File: 2020/Day 02/P1.rb
data = File.read("input.txt").split("\n")
total = 0

for line in data
    fLine = line.split(" ")
    tokenCount = 0

    range = fLine[0].split("-").map{|x| Integer(x)}
    token = fLine[1][0]
    password = fLine[2]

    for char in password.split("")
        if char === token
            tokenCount += 1
        end
    end

    if tokenCount >= range[0] && tokenCount <= range[1]
        total += 1
    end
end

puts "Result: #{total}"

File: src/2021/Day_10/P1.js
const fs = require(`fs`);
const path = require(`path`);

const pts = {
    [`)`]: 3,
    [`]`]: 57,
    [`}`]: 1197,
    [`>`]: 25137
};

const pairs = {
    [`(`]: `)`,
    [`[`]: `]`,
    [`{`]: `}`,
    [`<`]: `>`
};

const openings = Object.keys(pairs);

const errs = {
    [`)`]: 0,
    [`]`]: 0,
    [`}`]: 0,
    [`>`]: 0
};

const main = async () => {
    fs.readFile(path.resolve(__dirname, `./input.txt`), `utf-8`, (err, data) => {
        if (err) throw err;

        const lines = data.split(`\n`).map(x => x.trim());
        for (const line of lines) {
            const opened = [];

            const tokens = line.split(``);
            for (const token of tokens) {
                if (openings.includes(token)) opened.push(token);
                else if (pairs[opened.pop()] !== token) {
                    errs[token]++;
                    break;
                }
            }
        }

        let total = 0;
        for (const [token, count] of Object.entries(errs)) total += pts[token] * count;

        console.log(`Result: ${total}`);
    });
};

void main();
